assertk/src/commonMain/kotlin/assertk/assertions/list.kt

```diff
@@ -46,4 +49,44 @@ private fun List<*>.contentEquals(other: Array<*>): Boolean {
         if (get(i) != other[i]) return false
     }
     return true
-}
\ No newline at end of file
+}
+
+/**
+ * Asserts that a collection contains a subset of items the same order, but may have other items in the list.
+ *
+ * Usages:
+ *
+ * - `[]` containsAllInOrder `[1,2,3]` fails
+ * - `[1,2,3]` containsAllInOrder `[4,5,6]` fails
+ * - `[]` containsAllInOrder `[]` pass
+ * - `[1,2]` containsAllInOrder `[1,2,3]` pass
+ * - `[2,3,4]` containsAllInOrder `[1,2,3,4,5]` pass
+ *
+ * @param sublist The list of items it the actual list should contain in the same order.
+ */
+fun Assert<List<*>>.containsAllInOrder(sublist: List<*>) = given { actual: List<*> ->
+
+    if (sublist.isEmpty() && actual.isEmpty()) return@given
+
+    val sublistMatchCount =
+        when (val firstMatchOfSublist = if (sublist.isEmpty()) -1 else actual.indexOf(sublist.first())) {
+            -1 -> 0
+            else -> {
+                var n = 1
+                if (n < sublist.size) do {
+                    val a = actual[firstMatchOfSublist + n]
+                    val b = sublist[n]
+                    n += 1
+                } while (a == b && n < sublist.size && n < actual.size)
```

Comment:
If feel like this would be more clear as a while loop, thoughts?

```suggestion
                while (n < sublist.size && n < actual.size) {
                    val a = actual[firstMatchOfSublist + n]
                    val b = sublist[n]
                    if (a != b) break
                    n += 1
                }
```

Reply:
Thanx.  Much appreciated.  😄 👍